src/compiler/checker.ts

```diff
@@ -29418,26 +29418,27 @@ namespace ts {
                 (<PrefixUnaryExpression>expr).operand.kind === SyntaxKind.NumericLiteral;
         }
 
+        function checkAmbientInitializer(node: VariableDeclaration | PropertyDeclaration | PropertySignature) {
+            const isConstOrReadonly = isDeclarationReadonly(node) || isVariableDeclaration(node) && isVarConst(node);
+            if (node.initializer) {
+                if (isConstOrReadonly && !node.type) {
+                    if (!isStringOrNumberLiteralExpression(node.initializer)) {
+                        return grammarErrorOnNode(node.initializer, Diagnostics.A_const_initializer_in_an_ambient_context_must_be_a_string_or_numeric_literal);
+                    }
+                }
+                else {
+                    return grammarErrorOnNode(node.initializer, Diagnostics.Initializers_are_not_allowed_in_ambient_contexts);
+                }
+            }
+            if (node.initializer && !(isConstOrReadonly && isStringOrNumberLiteralExpression(node.initializer))) {

```

Comment:
I'm pretty sure you can move this inside the preceding `if`, as well as `const isConstOrReadonly = ...` above.

Then it would be worthwhile to invert the node.initializer check and return early.